lib/simplecov/result.rb

```diff
@@ -66,11 +66,15 @@ def to_hash
 
     # Loads a SimpleCov::Result#to_hash dump
     def self.from_hash(hash)
-      command_name, data = hash.first
-      result = SimpleCov::Result.new(data["coverage"])
-      result.command_name = command_name
-      result.created_at = Time.at(data["timestamp"])
-      result
+      results = []
+      hash.each do |command_name, data|
+        command_name, data = hash.first
```

Comment:
this seems odd to me, we iterate over it getting `command_name` and `data` and then we just discard them and use `hash.first` again - shouldn't this be prone to the same problem(s) again?